prelude/runtime/palloc.c

```diff
@@ -496,22 +497,23 @@ void sk_create_mapping(char* fileName, char* static_limit, size_t icapacity) {
     exit(ERROR_MAPPING_MEMORY);
   }
 
-  memcpy(persistent_fileName, fileName, fileName_length);
+  if (fileName != NULL) {
+    memcpy(persistent_fileName, fileName, fileName_length);
+  }
```

Comment:
Probably safer to init persistent_fileName to the empty string in the else case, even though it is not meaningful, it will stop string functions from walking memory if called without checking then enclosing ginfo->fileName for NULL.

Reply:
I didn't mean to change this diff's way of setting ginfo->fileName, just to change the hopefully-dead value in persistent_fileName when ginfo->fileName is NULL from something uninit to 0.